Don't commit assets

smacker opened the pull request:
fix #192

Now it works the same as any other apps project.
I also replaced bindata with esc.
`static_files_dev.go` & `static_files.go` are 100% copy-paste from lookout-web.

Makefile still looks a bit strange but changing it is out of the scope.

carlosms commented:
It's tested this locally, and for me `make build` & `make serve` fail if the dir `./server/asset` does not exist.
When the dir exists, `make serve` works, but `make packages` creates a bin serving the 501 errors.

smacker commented:
thanks for testing! I'll re-test with a clean state.

smacker commented:
good idea! 👍

smacker commented:
1. I added `mkdir -p` to make sure directory always exists
2. I can't reproduce the problem with `make packages`. This command and `make build` on clean repository generate correct binary with assets included.
3. I added simple integration test with check for correct content in `/` path
4. Had to rebase because `asset.go` was changed in master

dpordomingo commented:
pretty good idea adding integration tests ! :tada:

smacker commented on Makefile:
```diff
@@ -2,31 +2,42 @@
+GO_BUILD_ARGS := -ldflags "$(LD_FLAGS)" -tags "$(WITH_STATIC_TAG)"
+
+# Environment and arguments to use in `go run` calls.
+GO_RUN_ARGS += -tags "$(WITH_STATIC_TAG)"
```
in lookout-web it looks like this: https://github.com/src-d/lookout/blob/master/Makefile.web#L69
same-same

dpordomingo replied:
kind of same functionality, but subtle different code (ie. `GO_RUN_ARGS` does not appear in that `Makefile.web`)
But as I said, not blocker at all, so we can merge as it is.
(I also locally tested, and worked like a charm)

smacker replied:
yeah. Maybe we can actually extract go-handler part & Makefile somewhere and then just re-use in our projects. @carlosms wdyt? Shall I create an issue for that?

dpordomingo replied:
+1
I've been proposing that since years :tada:

carlosms replied:
sounds good to me. Let's open that issue :)

smacker replied:
https://github.com/src-d/feature-idea/issues/129

carlosms commented on server/cmd/bblfsh-web/main.go:
```diff
@@ -55,27 +51,12 @@ func main() {
-	// we handle urls on frontend
-	r.NoRoute(func(c *gin.Context) { c.File(indexPath) })
+
+	// TODO(@smacker): add configuration for ServerURL and FooterHTML
```
This is functionality brought from the code c&p from lookout, but we can't use yet in bblfsh web, right?
Should we have an issue for this, to implement it later?

smacker replied:
https://github.com/bblfsh/web/issues/201
